ethereum/contracts/pyth/PythGovernance.sol

```diff
@@ -67,13 +70,33 @@ abstract contract PythGovernance is PythGetters, PythSetters, PythGovernanceInst
 
     function upgradeUpgradableContract(UpgradeContractPayload memory payload) virtual internal;
 
-    function setGovernanceDataSource(bytes memory encodedPayload) internal {
-        SetGovernanceDataSourcePayload memory payload = parseSetGovernanceDataSourcePayload(encodedPayload);
-
+    // Transfer the governance data source to a new value with sanity checks
+    // to ensure the new governance data source can manage the contract.
+    function transferGovernanceDataSource(bytes memory encodedPayload) internal {
         PythInternalStructs.DataSource memory oldGovernanceDatSource = governanceDataSource();
 
-        setGovernanceDataSource(payload.newGovernanceDataSource);
-        setLastExecutedGovernanceSequence(payload.initialSequence);
+        TransferGovernanceDataSourcePayload memory payload = parseTransferGovernanceDataSourcePayload(encodedPayload);
+
+        // Make sure the VAA is valid
+        (IWormhole.VM memory vm, bool valid, string memory reason) = wormhole().parseAndVerifyVM(payload.claimVaa);
+        require(valid, reason);
+
+        GovernanceInstruction memory gi = parseGovernanceInstruction(vm.payload);
+        require(gi.targetChainId == 0, "governance data source change claim vaa should be applied to all chains");
```

Comment:
why should we care about this targetChainId at all? 

also please document all the requirements for creating a proper set of governance transfer messages above the struct definitions for their payloads.

Reply:
Explained my reasoning in another comment. Will decide based on your answer.

I'll change the js sdk so this is default to 0, so no documentation is needed anymore.